uportal-war/src/main/java/org/jasig/portal/portlets/tenantmanager/TenantManagerController.java

```diff
@@ -78,13 +121,55 @@ public void doAddTenant(ActionRequest req, @RequestParam("name") String name,
             }
         }
 
-        tenantService.createTenant(name, fname, attributes);
+        List<TenantOperationResponse> responses = new ArrayList<>();
+        tenantService.createTenant(name, fname, attributes, responses);
+
+        // Need to store some items to share with user in the report;  would be
+        // handy to have support for javax.portlet.actionScopedRequestAttributes
+        session.setAttribute(OPERATION_NAME_CODE, "tenant.manager.add");
+        session.setAttribute(OPERATIONS_LINTENER_RESPONSES, responses);
+
+        // Send the user to the report screen
+        res.setRenderParameter("action", "showReport");
 
     }
 
     @ActionMapping(params="action=doRemoveTenant")
-    public void doRemoveTenant(ActionRequest req, @RequestParam("fname") String fname) {
-        tenantService.deleteTenantByFName(fname);
+    public void doRemoveTenant(ActionRequest req, ActionResponse res,
+            final PortletSession session, @RequestParam("fname") String fname) {
+
+        List<TenantOperationResponse> responses = new ArrayList<>();
+        tenantService.deleteTenantByFName(fname, responses);
+
+        // Need to store some items to share with user in the report;  would be
+        // handy to have support for javax.portlet.actionScopedRequestAttributes
+        session.setAttribute(OPERATION_NAME_CODE, "tenant.manager.remove.tenant");
+        session.setAttribute(OPERATIONS_LINTENER_RESPONSES, responses);
+
+        // Send the user to the report screen
+        res.setRenderParameter("action", "showReport");
+
+    }
+
+    @ActionMapping(params="action=doListenerAction")
+    public void doListenerAction(ActionRequest req, ActionResponse res,
+            @RequestParam("fname") String fname, final PortletSession session) {
+
+        final ITenantManagementAction action = tenantService.getAction(fname);
+        final ITenant tenant = (ITenant) session.getAttribute(CURRENT_TENANT_SESSION_ATTRIBUTE);
+        if (tenant == null) {
+            throw new IllegalStateException("No current tenant");
+        }
+        TenantOperationResponse response = action.invoke(tenant);
+
+        // Need to store some items to share with user in the report;  would be
+        // handy to have support for javax.portlet.actionScopedRequestAttributes
+        session.setAttribute(OPERATION_NAME_CODE, action.getMessageCode());
+        session.setAttribute(OPERATIONS_LINTENER_RESPONSES, Collections.singletonList(response));
+
+        // Send the user to the report screen
+        res.setRenderParameter("action", "showReport");
```

Comment:
DRY Principal:  refactor session manipulation and FWD to showReport to a private method.